dashboard/test/models/contact_rollups_raw_test.rb

```diff
@@ -30,14 +43,37 @@ class ContactRollupsRawTest < ActiveSupport::TestCase
     refute_nil ContactRollupsRaw.find_by(email: teacher.email, data: nil, sources: 'dashboard.users')
   end
 
+  test 'extract_from_source_query can import when source is a subquery' do
+    first_child = create :student, parent_email: 'caring@parent.com'
+    second_child = create :student, parent_email: 'caring@parent.com'
+
+    # we're not actually interested in user IDs in contact rollups
+    # just a simple example of something we could extract in a subquery
+    subquery = <<~SQL
+      SELECT parent_email, max(updated_at) as updated_at, max(id) as higher_student_id
```

Comment:
I'm not sure why this subquery has to be complicated? Any subquery would do.

Reply:
I just wanted something that involved some sort of aggregation in the subquery that wasn't explicitly the one that is required for the parent email. Can update though.

Reply:
I'd prefer keeping the subquery simple. We just want to assure that the method-under-test can process subquery. If the test fail, it's easier to isolate the issue if there are less complicated/moving parts.

Reply:
I'm not sure how to make it simpler -- the simplest case where we'd use a subquery is when email in a given source isn't unique, which would require an aggregation on the `updated_at` and some other piece of data that we're trying to extract (modeled here as the maximum ID). Do you have an idea?

Reply:
Ah I see your point. You're testing not only `get_extraction_query` can generate the correct query but also the result of executing that query is correct. It's good then.